src/box/memtx_hash.cc

```diff
@@ -582,6 +582,9 @@ hash_read_view_iterator_next_raw(struct index_read_view_iterator *iterator,
 			*data = NULL;
 			return 0;
 		}
+		if (rv->base.filter_tuple &&
+		    !rv->base.filter_tuple(*res, rv->base.def))
+			continue;
```

Comment:
Why do you hide those tuples from read-views? They are still visible in normal selects. What is the point? It just makes read-views inconsistent with normal unprotected reading.

If you are doing it for the snapshot, I suggest to patch just the snapshotting code. Not the read-views.

Reply:
Made modifications: extracted `filter_tuple` feature into a preparatory commit, and only set it when creating a read_view for checkpointing and replica joining.

However this particular part didn't change, because snapshotting code still goes through here

Reply:
The comment wasn't about extracting it though. You don't need this to be a part of the read views at all. This is solely for snapshots. Then just change the snapshot code. It is located in the single file - memtx_engine.cc, one function - `checkpoint_f()`. Filter the tuples returned from the read view. Not the read view internals.

Reply:
done